### PR TITLE
x11-misc/xkblayout-state: initial commit

### PR DESCRIPTION
**Pull request by karllinden**
`xkblayout-state` is a small command-line program to get/set the current XKB keyboard layout. I use it to show the current keyboard layout on my laptop (which has `us` keyboard layout, but sometimes i need to switch to `se`).

This package is available in ArchLinux's AUR, so I think it is appropriate to have it in Gentoo too.

Since upstream seems to do releases via GitHub and because those releases tend to change checksum frequently I have not made a versioned ebuild. Therefore this PR only features a live ebuild. Because the project has not seen many changes lately (except some Makefile changes I wrote to make the ebuild compile without patches) it might make sense to make a snapshot and place somewhere.

### CODE REVIEW

**Comment by karllinden** [comment]
@monsieurp I have added a snapshot version of the current HEAD and added it to the initial commit. As a consequence of the versioning I changed the live ebuild to be version 99999999.

File: x11-misc/xkblayout-state/metadata.xml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+	<maintainer type="person">
+		<email>karl.j.linden@gmail.com</email>
+		<name>Karl Linden</name>
+	</maintainer>
+	<maintainer type="project">
+	<email>proxy-maint@gentoo.org</email>
```
Comment (mgorny) [inline x11-misc/xkblayout-state/metadata.xml]: Misindent.

File: x11-misc/xkblayout-state/xkblayout-state-20171214.ebuild [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit toolchain-funcs
+
+# Commit Date: 27 Nov 2017
+COMMIT="d023bd7f58a54da08c3aad3c234a629b8a010892"
```
Comment (mgorny) [inline x11-misc/xkblayout-state/xkblayout-state-20171214.ebuild]: I'd suggest naming it EGIT_COMMIT for consistency.

File: x11-misc/xkblayout-state/xkblayout-state-20171214.ebuild [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+DESCRIPTION="A small program to get/set the current XKB layout"
+HOMEPAGE="https://github.com/nonpop/xkblayout-state"
+SRC_URI="${HOMEPAGE}/archive/${COMMIT}.tar.gz -> ${P}.tar.gz"
```
Comment (mgorny) [inline x11-misc/xkblayout-state/xkblayout-state-20171214.ebuild]: Don't use HOMEPAGE here. It is multi-valued and you don't want the ebuild to fail randomly when someone adds second homepage.

File: x11-misc/xkblayout-state/xkblayout-state-20171214.ebuild [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+KEYWORDS="~amd64 ~x86"
+IUSE=""
+
+S="${WORKDIR}/${PN}-${COMMIT}"
+
+RDEPEND="x11-libs/libX11"
+DEPEND="${RDEPEND}"
+
+src_compile() {
+	emake CXX=$(tc-getCXX)
```
Comment (mgorny) [inline x11-misc/xkblayout-state/xkblayout-state-20171214.ebuild]: Quote it, tc-getCXX can return string with spaces.

File: x11-misc/xkblayout-state/xkblayout-state-20171214.ebuild [hunk 5]
```diff
@@ -0,0 +1,31 @@
+# Copyright 1999-2017 Gentoo Foundation
```
Comment (mgorny) [inline x11-misc/xkblayout-state/xkblayout-state-20171214.ebuild]: Please don't use such long versions, use `0_p20171214`.

**Comment by mgorny** [comment]
Ping. You still there?

**Comment by karllinden** [comment]
@mgorny I'm still here. I will take care of your comments when I find the time to do it.

**Comment by mgorny** [comment]
Ok, thanks for getting back to me.

File: x11-misc/xkblayout-state/xkblayout-state-20171214.ebuild (hunk 5 above)
Comment (karllinden) [reply x11-misc/xkblayout-state/xkblayout-state-20171214.ebuild]: Done.

File: x11-misc/xkblayout-state/xkblayout-state-20171214.ebuild (hunk 4 above)
Comment (karllinden) [reply x11-misc/xkblayout-state/xkblayout-state-20171214.ebuild]: Done.

File: x11-misc/xkblayout-state/xkblayout-state-20171214.ebuild (hunk 3 above)
Comment (karllinden) [reply x11-misc/xkblayout-state/xkblayout-state-20171214.ebuild]: Done.

File: x11-misc/xkblayout-state/xkblayout-state-20171214.ebuild (hunk 2 above)
Comment (karllinden) [reply x11-misc/xkblayout-state/xkblayout-state-20171214.ebuild]: Done.

File: x11-misc/xkblayout-state/metadata.xml (hunk 1 above)
Comment (karllinden) [reply x11-misc/xkblayout-state/metadata.xml]: Done.

**Comment by karllinden** [comment]
@mgorny I updated the ebuilds in accordance with your comments. Copyright updated and branch rebased.

**Comment by mgorny** [comment]
Looks good now. Thanks!

**Pull request closed without merging**